[35mDEBUG   [0m selenium.webdriver.remote.remote_connection:remote_connection.py:313 POST http://127.0.0.1:4724/session {"capabilities": {"firstMatch": [{}], "alwaysMatch": {"platformName": "Android", "appium:deviceName": "SM-S921N", "appium:appPackage": "com.hyundai.shucle.gmaas", "appium:appActivity": "com.hyundai.airlab.shucle.presenter.MainActivity", "appium:automationName": "UiAutomator2", "appium:noReset": true, "appium:fullReset": false, "appium:newCommandTimeout": 300, "appium:uiautomator2ServerLaunchTimeout": 60000}}}
[35mDEBUG   [0m urllib3.connectionpool:connectionpool.py:241 Starting new HTTP connection (1): 127.0.0.1:4724
[35mDEBUG   [0m urllib3.connectionpool:connectionpool.py:544 http://127.0.0.1:4724 "POST /session HTTP/1.1" 500 0
[35mDEBUG   [0m selenium.webdriver.remote.remote_connection:remote_connection.py:329 Remote response: status=500 | data={"value":{"error":"unknown error","message":"Could not find a connected Android device in 20000ms","stacktrace":"UnknownError: Could not find a connected Android device in 20000ms\n    at getResponseForW3CError (C:\\Users\\tslee\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\protocol\\errors.ts:1103:34)\n    at asyncHandler (C:\\Users\\tslee\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\@appium\\base-driver\\lib\\protocol\\protocol.ts:507:57)\nThe above error is caused by\nError: Could not find a connected Android device in 20000ms\n    at ADB.getDevicesWithRetry (C:\\Users\\tslee\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\appium-adb\\lib\\tools\\system-calls.js:295:13)\n    at AndroidUiautomator2Driver.getDeviceInfoFromCaps (C:\\Users\\tslee\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\appium-android-driver\\lib\\commands\\device\\common.js:41:21)\n    at AndroidUiautomator2Driver.createSession (C:\\Users\\tslee\\.appium\\node_modules\\appium-uiautomator2-driver\\lib\\driver.ts:361:30)\n    at AppiumDriver.createSession (C:\\Users\\tslee\\AppData\\Roaming\\npm\\node_modules\\appium\\lib\\appium.js:360:35)"}} | headers=HTTPHeaderDict({'X-Powered-By': 'Express', 'Vary': 'X-HTTP-Method-Override', 'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '1224', 'ETag': 'W/"4c8-V6LzNuVgks5H208V9v99zXinryA"', 'Date': 'Sun, 19 Oct 2025 13:13:39 GMT', 'Connection': 'keep-alive', 'Keep-Alive': 'timeout=600'})
[35mDEBUG   [0m selenium.webdriver.remote.remote_connection:remote_connection.py:356 Finished Request